frontend/src/components/activePage/NestedTables/NestedTables.tsx

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+import { FC, useState } from 'react';
+import { Table } from 'antd';
+import { CaretRightOutlined } from '@ant-design/icons';
+import RowHeading from '../RowHeading/RowHeading';
+
+export interface IWorkspace {
+  key: string;
+  id: string;
+  name: string;
+  templates: Array<ITemplate>;
+}
+
+export interface ITemplate {
+  key: string;
+  id: string;
+  name: string;
+  workspace: string;
+  nActiveInstances: number;
+}
+export interface INestedTablesProps {
+  workspaces: Array<IWorkspace>;
+  templates: Array<ITemplate>;
+  isManager: boolean;
+  destroyAll: () => void;
+}
+
+type rowType = IWorkspace | ITemplate;
+
+const NestedTables: FC<INestedTablesProps> = ({ ...props }) => {
+  const { workspaces, templates, isManager, destroyAll } = props;
+  const { Column } = Table;
+  const [expandedID, setExpandedID] = useState(['']);
+
+  const accordion = (expanded: boolean, record: rowType) => {
+    isManager
+      ? expanded
+        ? setExpandedID([(record as IWorkspace).key])
+        : setExpandedID([''])
+      : expanded
+      ? setExpandedID([(record as ITemplate).key])
+      : setExpandedID(['']);
+  };
```

Comment:
```suggestion
  const accordion = (expanded: boolean, record: rowType) => {
    const expId = !expanded ? '' 
      : (isManager ? (record as IWorkspace).key : (record as ITemplate).key);
    setExpandedID([expId]);
  };
```
This part seems a bit redundant and not too easy to be read. 
I tried to make it cleaner but I don't know if I succeeded. You're totally free to reject this suggestion (especially if it's not correct 😂).

Reply:
Oh yes, much better thanks!